tests/ha/haproxy.pm

```diff
@@ -0,0 +1,107 @@
+# SUSE's openQA tests
+#
+# Copyright (c) 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test haproxy resource agent
+# Maintainer: Julien Adamek <jadamek@suse.com>
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use lockapi;
+use utils 'zypper_call';
+use utils 'systemctl';
+use hacluster;
+
+sub run {
+    my $cluster_name = get_cluster_name;
+    my $haproxy_rsc  = 'haproxy';
+    my $haproxy_cfg  = '/etc/haproxy/haproxy.cfg';
+    my $apache_file  = '/srv/www/htdocs/index.html';
+    my $vip_ip       = '10.0.2.20';
+    my $vip_rsc      = 'vip';
+    my $node_01      = choose_node(1);
+    my $node_02      = choose_node(2);
+    my $node_01_ip   = script_output "host -t A $node_01 | awk '{ print \$NF }'";
+    my $node_02_ip   = script_output "host -t A $node_02 | awk '{ print \$NF }'";
+
+    # Waiting for the other nodes to be ready
+    barrier_wait("HAPROXY_INIT_$cluster_name");
+
+    # Installation of haproxy and apache2 packages
+    zypper_call 'in haproxy apache2';
+    save_screenshot;
+
+    # Get apache file template from the openQA server
+    assert_script_run "curl -f -v " . autoinst_url . "/data/ha/haproxy_apache.template -o $apache_file";
+
+    if (is_node(1)) {
+        # Get haproxy configuration file template from the openQA server
+        assert_script_run "curl -f -v " . autoinst_url . "/data/ha/haproxy.cfg.template -o $haproxy_cfg";
+
+        # And modify the template according to our needs
+        assert_script_run "sed -i 's/%NODE_01%/$node_01/g' $haproxy_cfg";
+        assert_script_run "sed -i 's/%NODE_01_IP%/$node_01_ip/g' $haproxy_cfg";
+        assert_script_run "sed -i 's/%NODE_02%/$node_02/g' $haproxy_cfg";
+        assert_script_run "sed -i 's/%NODE_02_IP%/$node_02_ip/g' $haproxy_cfg";
+        assert_script_run "sed -i 's/%VIP_IP%/$vip_ip/g' $haproxy_cfg";
+
+        add_file_in_csync(value => "$haproxy_cfg");
+
+        # Execute csync2 to synchronise the configuration files
+        exec_csync;
+
+        # Modify the apache template file according to our needs
+        assert_script_run "sed -i 's/%NODE%/$node_01/g' $apache_file";
+        assert_script_run "sed -i 's/%IP%/$node_01_ip/g' $apache_file";
+    }
+    elsif (is_node(2)) {
+        # Modify the apache template file according to our needs
+        assert_script_run "sed -i 's/%NODE%/$node_02/g' $apache_file";
+        assert_script_run "sed -i 's/%IP%/$node_02_ip/g' $apache_file";
+    }
+
+    # Apache have to be started in the both nodes for load balancing
+    systemctl 'enable --now apache2';
```

Comment:
Looks OK for the first version, but maybe better to add Apache2 in the HA configuration for a next version, no? (I must admit that I didn't checked the HAProxy best practices for this).

Reply:
Good point, we will benefit to test Apache2 RA as well.
For me, it's definitely not a best practice to use haproxy and apache2 on the same nodes on production, but it makes sense to do it as we are doing test.
I could do it after the first version.